pkg/controller/controllerutil/controllerutil.go

```diff
@@ -168,3 +169,61 @@ func mutate(f MutateFn, key client.ObjectKey, obj runtime.Object) error {
 
 // MutateFn is a function which mutates the existing object into it's desired state.
 type MutateFn func() error
+
+// AddFinalizer accepts a metav1 object and adds the provided finalizer if not present.
+func AddFinalizer(o metav1.Object, finalizer string) {
+	newFinalizers := AddString(o.GetFinalizers(), finalizer)
+	o.SetFinalizers(append(newFinalizers, finalizer))
```

Comment:
huh?  Doesn't this add the finalizer twice?

Reply:
pushed early, fixed